Underscore: various small fixes

webbiesdk opened the pull request:
A bunch of fixes, and added a lot of "or undefined" for strict nulls (by no means all of them, there is still a LOT of methods that can do something unexpected under strict-nulls).

Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present. (Reports a lot of stuff, but it also did before my changes)

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>> (not relevant)
- [x] Increase the version number in the header if appropriate. (But it shouldn't be, I've updated the version number once before: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/11117)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

minestarks commented:
@webbiesdk Could you turn on ```strictNullChecks``` in the ```tsconfig.json``` and update the tests to protect against further strict null mistakes?

webbiesdk commented:
> to protect against further strict null mistakes
I want to quote [Dijkstra](https://www.brainyquote.com/quotes/quotes/e/edsgerdijk201165.html).  

I changed `strictNullChecks` to true, and updated the tests to reflect that (only a few changes needed).

minestarks commented:
> If a test could have prevented this bug, it will likely prevent future bugs too. - @minestarks 

Looks good, thanks!
